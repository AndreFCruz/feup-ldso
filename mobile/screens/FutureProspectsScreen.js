import React from 'react';
import { StyleSheet, Image } from 'react-native';
import { Container, Content, View, Text, H1 } from 'native-base';
import { connect } from 'react-redux';
import { getFutureProspects } from '../reducers/modules/facultyReducer';

class FutureProspectsScreen extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      banner: require('../assets/images/banners/feup-banner.png')
    };
  }

  componentDidMount() {
    const { language, name } = this.props;
    
    this.props.getFutureProspects(language, name);
  }

  componentDidUpdate(prevProps) {
    const { language, name } = this.props;

    if (prevProps.language != language)
      this.props.getFutureProspects(language, name);
  }

  render() {
    const { name, loading, prospects, language } = this.props;

    if (loading) {
      return (
        <View>
          <Text>{ language == 'en'? 'Loading...' : 'Carregando...' }</Text>
        </View>
      );
    }
    return (
      <Container style={styles.container}>
        <Content contentContainerStyle={styles.content} >
          <H1>{ language == 'en'? 'What is ' + name + ' planning?' : 'O que est√° a ' + name + ' a planear?'}</H1>

          <Image
            source={this.state.banner}
            style={styles.images}
          />

          <Text style={styles.text}>
            {prospects}
          </Text>
        </Content>
      </Container>
    );
  }

}


const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  content: {
    paddingTop: 50,
    alignItems: 'center'
  },
  images: {
    marginTop: 30,
  },
  text: {
    paddingTop: 30,
    paddingBottom: 30,
    paddingRight: 10,
    paddingLeft: 10,
    maxWidth: 500,
    textAlign: 'justify'
  }
});

const mapStateToProps = ({ faculty, language }) => ({
  name: faculty.name,
  loading: faculty.loading,
  prospects: faculty.futureProspects,
  language: language.selection
});

const mapDispatchToProps = {
  getFutureProspects
};

export default connect(mapStateToProps, mapDispatchToProps)(FutureProspectsScreen);