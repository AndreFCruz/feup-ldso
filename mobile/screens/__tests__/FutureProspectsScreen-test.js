import 'react-native';
import React from 'react';
import FutureProspectsScreen from '../FutureProspectsScreen';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

function setup(loading = false) {

    const initialState = { 
        faculty: {
            loading: loading,
            futureProspects: {
                'content': '[EN] Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc sem eros, elementum sed dui et, tincidunt hendrerit mauris. Nunc vestibulum massa eget condimentum aliquam. Sed in convallis nulla. Duis aliquam lorem tempor arcu dapibus, non laoreet nisi bibendum. Nulla nec libero augue. Pellentesque scelerisque lectus non diam molestie fringilla. Vestibulum a eros justo. Sed tincidunt quam eu dui auctor, sit amet fringilla tellus ultricies. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Aliquam sit amet varius lacus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Sed tortor justo, ornare a vulputate in, mattis id mi. Donec varius, dui venenatis pharetra ultricies, ligula nulla ornare ex, quis mattis libero quam a tellus. Quisque quis est id nulla ornare semper at a nisi. Integer tincidunt egestas ornare. Nullam congue tellus arcu, eu ultricies mi pellentesque eget. Pellentesque massa sapien, suscipit sit amet lacus vel, faucibus faucibus elit. Cras in metus viverra, luctus magna ac, tristique leo. Aenean ultrices dui eget velit cursus, vitae suscipit purus varius. Aliquam erat volutpat. In porta metus et velit aliquet porta. Sed tempor dolor bibendum pretium euismod. Etiam eros mi, feugiat aliquet mattis non, elementum nec sem. Aliquam eget erat porta, porta risus nec, placerat orci. Integer nec enim tempor orci mollis luctus non at nisi. Maecenas fermentum magna tempor nisl rutrum tempor. Quisque in rutrum velit, ut finibus risus. Morbi pharetra placerat neque efficitur efficitur.',
                'banner': 'http://www.construir.pt/wp-content/uploads/2017/10/feup-1024x624.jpg'
            },
        },
        language: {
            selection: 'en'
        }
    };

    const wrapper = shallow(<FutureProspectsScreen />,
        {
            context: { store: mockStore(initialState) },
            disableLifecycleMethods: true
        });

    return wrapper;
}

it('renders loading screen', () => {
    const wrapper = setup(true);

    expect(wrapper.dive()).toMatchSnapshot();
});
    
it('renders screen with future prospect content', () => {
    const wrapper = setup();

    expect(wrapper.dive()).toMatchSnapshot();
});