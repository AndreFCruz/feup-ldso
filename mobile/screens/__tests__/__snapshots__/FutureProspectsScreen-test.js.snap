// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading screen 1`] = `
<Styled(ViewNB)>
  <Styled(Text)>
    Loading...
  </Styled(Text)>
</Styled(ViewNB)>
`;

exports[`renders screen with future prospect content 1`] = `
<Styled(Container)
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Styled(Content)
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "paddingTop": 50,
      }
    }
  >
    <Styled(H1)>
      What is undefined planning?
    </Styled(H1)>
    <Image
      source={
        Object {
          "uri": "http://www.construir.pt/wp-content/uploads/2017/10/feup-1024x624.jpg",
        }
      }
      style={
        Object {
          "height": 200,
          "marginTop": 30,
          "width": 300,
        }
      }
    />
    <Styled(Text)
      style={
        Object {
          "maxWidth": 500,
          "paddingBottom": 30,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 30,
          "textAlign": "justify",
        }
      }
    >
      [EN] Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc sem eros, elementum sed dui et, tincidunt hendrerit mauris. Nunc vestibulum massa eget condimentum aliquam. Sed in convallis nulla. Duis aliquam lorem tempor arcu dapibus, non laoreet nisi bibendum. Nulla nec libero augue. Pellentesque scelerisque lectus non diam molestie fringilla. Vestibulum a eros justo. Sed tincidunt quam eu dui auctor, sit amet fringilla tellus ultricies. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Aliquam sit amet varius lacus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Sed tortor justo, ornare a vulputate in, mattis id mi. Donec varius, dui venenatis pharetra ultricies, ligula nulla ornare ex, quis mattis libero quam a tellus. Quisque quis est id nulla ornare semper at a nisi. Integer tincidunt egestas ornare. Nullam congue tellus arcu, eu ultricies mi pellentesque eget. Pellentesque massa sapien, suscipit sit amet lacus vel, faucibus faucibus elit. Cras in metus viverra, luctus magna ac, tristique leo. Aenean ultrices dui eget velit cursus, vitae suscipit purus varius. Aliquam erat volutpat. In porta metus et velit aliquet porta. Sed tempor dolor bibendum pretium euismod. Etiam eros mi, feugiat aliquet mattis non, elementum nec sem. Aliquam eget erat porta, porta risus nec, placerat orci. Integer nec enim tempor orci mollis luctus non at nisi. Maecenas fermentum magna tempor nisl rutrum tempor. Quisque in rutrum velit, ut finibus risus. Morbi pharetra placerat neque efficitur efficitur.
    </Styled(Text)>
  </Styled(Content)>
</Styled(Container)>
`;
